keps/sig-api-machinery/20190228-priority-and-fairness.md

```diff
@@ -777,127 +788,147 @@ kind: FlowSchema
-    - superSet:
-      field: groups
-      set: [ "system:masters" ]
+  matchingPrecedence: 1500
+  priorityLevelConfiguration:
+    name: exempt
+  rules:
+  - rule:
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:masters"
+  - rule:
+      verbs: ['*']
+      apiGroups: ['authentication.k8s.io', 'authorization.k8s.io']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:authenticated"     
+    - kind: Group
+      name: "system:unauthenticated"     
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
   name: system-high
 spec:
-  requestPriority:
+  matchingPrecedence: 2500
+  priorityLevelConfiguration:
     name: system-high
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # heartbeats by nodes
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: resource
-      value: nodes
-  - and: # kubelet and kube-proxy ops on system objects
-    - superSet:
-      field: groups
-      set: [ "system:nodes" ]
-    - equals:
-      field: namespace
-      value: kube-system
-  - and: # leader elections for system controllers
-    - patternMatch:
-      field: user
-      pattern: system:controller:.*
-    - inSet:
-      field: resource
-      set: [ "endpoints", "configmaps", "leases" ]
-    - equals:
-      field: namespace
-      value: kube-system
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule:
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: Group
+      name: "system:nodes"
+  - rule:
+      verbs: ['*']
+      nonResourceURLs: ['*']
+    subjects:
+    - kind: Group
+      name: "system:nodes"    
+  - rule:
+      verbs: ['*']
+      apiGroups: ['coordination.k8s.io']
+      resources: ['leases']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler" 
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
   name: system-low
 spec:
-  matchingPriority: 900
-  requestPriority:
+  matchingPrecedence: 3500
+  priorityLevelConfiguration:
     name: system-low
-  flowDistinguisher:
-    source: user
-    # no transformation in this case
-  match:
-  - and: # the garbage collector
-    - equals:
-      field: user
-      value: system:controller:garbage-collector
+  distinguisherMethod:
+    type: ByUser
+  rules:  
+  - rule:
+      verbs: ['*']
+      apiGroups: ['*']
+      resources: ['*']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler"    
+  - rule:
+      verbs: ['*']
+      nonResourceURLs: ['*']
+    subjects:
+    - kind: User
+      name: "system:kube-controller-manager"    
+    - kind: User
+      name: "system:kube-scheduler"         
 ```
 
 ```yaml
 kind: FlowSchema
 meta:
-  name: workload-high
+  name: workload
```

Comment:
The original design was that the workload consists of two parts: kubectl requests from users and requests from controller.  The original design prioritized kubectl requests higher than controller requests.  The revised design prioritizes the controller requests higher (in "system-low"); I think this is not an improvement.

Reply:
i think it's better we keep it simple in the alpha stage. users can customize their own settings theirself?

Reply:
This is not really a matter of simplicity.  We already made an estimate of what we think will be a good configuration.  I do not think it is an improvement to switch to a less good configuration just because it can be overridden.